www/waterfall_view/src/module/main.module.js

```diff
@@ -635,14 +644,89 @@ var WaterfallController = (function() {
             });
         }
 
-        renderNewData() {
+        toggleTag(tag) {
+            if (!this.$scope.tags_filter.includes(tag)) {
+                this.$scope.tags_filter.push(tag);
+            } else {
+                this.$scope.tags_filter = _.remove(this.$scope.tags_filter, function(currentTag) {
+                    return currentTag != tag;
+                });
+            }
+            this.$location.search("tags", this.$scope.tags_filter);
+        }
+
+        isTagFiltered(tag) {
+            if (this.$scope.tags_filter.includes(tag)) {
+                return true;
+            } else {
+                return false;
+            }
+        }
+
+        clearTags() {
+            this.$scope.tags_filter = this.tags_filter = [];
+            this.$location.search("tags", this.$scope.tags_filter);
+        }
+
+        makeTagBuilders(currentTags, builders) {
+            let tag_builders = [];
+            let anyTagSelected = false;
+            if (typeof currentTags != 'string') {
+                anyTagSelected = true;
+                for (const builder of builders) {
+                    let v = currentTags.every(currentTag =>
```

Comment:
The every is actually weird for me, after testing. I would rather expect any, so that you can viiew several tag at the same time.

Reply:
It does work with several tags at the same time

Reply:
I saw that you offer three states of tags for builders: required (`and`), forbidden (`and not`), "optional" (`or`).
My ideal scenario (but not super easy to implement) might be:
* If the tag is required (with a `+`) or forbidden, hide all other tags which would result in an empty set of builders being shown.
* If a tag is optional, still show all other tags.
I suggested to ask you for opinion, but I still felt that some basic (but working) functionality would be more helpful than trying and failing to cover all the possible complex scenarios?

Reply:
Both `and` and `or` are valid requests and there are use cases for both. Say that you want to see both builders on Mac OS X 10.6 building wine (32-bit and 64-bit). Then `or` is not all that helpful, you need `+wine +10.6` (`'wine' and '10.6'`); `('wine' or '10.6')` would be pretty useless.

On the other hand you may want to see one port building on 10.8 and 10.9. So you would need something like `+wine 10.8 10.9` (`'wine' and ('10.8' or '10.9')`).

I don't want to argue too much for either option or either complexity of implementation. I leave it up to developers to say what they want to see implemented. Even what's there now is more helpful than no tags (the only confusing thing is that you currently all to easily end up with an empty set of builders after clicking two tags).

Reply:
Yes that would be really cool. I will implement it if @tardyp is fine with this.